### PR TITLE
Add subgroups visibility field, take 2

### PR DESCRIPTION
**Pull request by simonlindholm**
Replaces #83. This PR is mostly just a proof-of-concept for the idea of warning for confusing sort order. It's not ready to merge; still waiting on @niemela to comment on what flag name/default to use, and to update the spec.

### CODE REVIEW

File: problemtools/verifyproblem.py [hunk 1]
```diff
@@ -549,6 +583,11 @@ def check(self, args):
         elif self._data['license'] == 'unknown':
             self.warning("License is 'unknown'")
 
+        if self._data['grading']['testgroups'] not in [True, False]:
```
Comment (niemela) [inline problemtools/verifyproblem.py]: Change key name to `test_data_groups`.

File: problemtools/verifyproblem.py [hunk 2]
```diff
@@ -549,6 +583,11 @@ def check(self, args):
...
+            self.error("Invalid value for grading.testgroups: %s" % self._data['grading']['testgroups'])
+        elif self._data['grading']['testgroups'] and self._data['type'] == 'pass-fail':
+            self.error("Test groups are only supported for scoring problems, this is a pass-fail problem")
```
Comment (niemela) [inline problemtools/verifyproblem.py]: The `grading` map can only be used with scoring problems, so this check should be one level higher.

**Comment by simonlindholm** [comment]
> Please comment on the name chosen (test_data_groups).

I'd prefer `testgroups`, but it's not a big deal. (Reasoning: either way it's confusing -- the flag doesn't turn on test data groups but merely displays them -- but I can't come up with a much better term. "testgroups" is a bit more informal, familiar and sounds better, and perhaps ties a bit closer to the UI concept than to the spec's "test data groups". Not that people care about formal spec terms. Which might also reduce the reason for using the spec term, in addition to it sounding odd.)

File: problemtools/verifyproblem.py (hunk 2 above)
Comment (niemela) [reply problemtools/verifyproblem.py]: Ok, fair enough...

**Comment by niemela** [comment]
> I'd prefer `testgroups`, but it's not a big deal. (Reasoning: either way it's confusing -- the flag doesn't turn on test data groups but merely displays them -- but I can't come up with a much better term. "testgroups" is a bit more informal, familiar and sounds better, and perhaps ties a bit closer to the UI concept than to the spec's "test data groups". Not that people care about formal spec terms. Which might also reduce the reason for using the spec term, in addition to it sounding odd.)

If the term ("Test Data Group") is not good we should update the spec, not just use different terms in different places. I *do* think that "Test Data Group" is better than "Test Group".

Regarding spaces or no spaces, there is a space in "Test Group" so there should be one in the key as well.

I still think `test_data_group` is better than `testgroup` (or `test_group`).

**Comment by simonlindholm** [comment]
Merged with that name change. Please update the wiki @niemela.

**Comment by simonlindholm** [comment]
> Please update the wiki @niemela.

Ping. (Not urgent, but we'll forget this at some point...)

**Comment by simonlindholm** [comment]
Ping @niemela...

**Comment by austrin** [comment]
@simonlindholm can you summarize what are the spec updates to be made and I can make them?

**Comment by simonlindholm** [comment]
`test_data_groups` should change to `show_test_data_groups`.

**Comment by austrin** [comment]
Ah ok it was that trivial.  Fixed now.

**Comment by simonlindholm** [comment]
Thanks!
